### DISCUSSION TITLE
The Europan economy and its relation to missions.

### DISCUSSION THREAD

**Discussion by Cogs16** [FakeFishGames/Barotrauma#5972] [Ideas and suggestions]
I understand this is a hope and dream suggestion, BUT HERE IT IS any way. It would be awesome if we could see a dynamic economy based on the supply and demand of supplies at each outpost. Additionally, workers and NPCs would also be in demand. The missions you take in-game such as personal transport or cargo are actually linked into this system. You would be delivering people and supplies that are actually needed by another outpost.

The system would work like this. Each outpost produces items based on specialty and demands the items it does not produce on its own. If there is high demand for certain items, this would influence what type of outpost a habitat turns into. Additionally, random simulated events may influence demand, such as a crazed mudraptor attack, causing the outpost to require lots of supplies and driving the price up at the outpost and perhaps even globally. If an outpost isn't supplied for too long, or doesn't receive new people to replace losses, it will eventually die. Habitation outposts, and the colony, would produce NPCs for other outposts to recieve.

This all leaves the question of how supplies will be transported while players are away from certain areas. I propose that in the future, there would be say 5 other simulated submarines on europa, which mainly do cargo missions. These subs locations would appear with a marker on the minimap. They would also move in a turn-based fashion. One day in the future, if submarine AI is implemented, these subs would actually be encounterable. Lastly, these subs would be influenced by swarms that would appear on the map and occasionally change positions. These swarms would be persistent and also move around in a turn based manner, making them feel more like real tangible parts of Europa. Subs that pass through swarms have a chance of being sunk, or having their cargo haul reduced. New subs would spawn from the colony and perhaps a shipyard outpost type. 

The aim of this system is to help players believe that they can make a REAL difference on europa, and to add another layer of complexity to the game.

**Comment by Veritius** [comment]
You might want to check out the [Roadmap](https://trello.com/b/hBXI8ltN/barotrauma-roadmap-known-issues) on Trello.

**Reply by Cogs16** [reply]
Can you point out to me where this is on the roadmap?
